impl/src/main/java/edu/arizona/rice/kew/docsearch/dao/impl/DocumentSearchResultsSizeDAOJdbcImpl.java

```diff
@@ -0,0 +1,99 @@
+package edu.arizona.rice.kew.docsearch.dao.impl;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.List;
+
+import javax.sql.DataSource;
+
+import org.apache.commons.lang.StringUtils;
+import org.kuali.rice.core.api.uif.RemotableAttributeField;
+import org.kuali.rice.kew.api.document.search.DocumentSearchCriteria;
+import org.kuali.rice.kew.docsearch.service.DocumentSearchService;
+import org.kuali.rice.kew.doctype.bo.DocumentType;
+import org.kuali.rice.kew.impl.document.search.DocumentSearchGenerator;
+import org.springframework.dao.DataAccessException;
+import org.springframework.jdbc.core.ConnectionCallback;
+import org.springframework.jdbc.core.JdbcTemplate;
+
+import edu.arizona.rice.kew.docsearch.dao.DocumentSearchResultsSizeDAO;
+
+
+public class DocumentSearchResultsSizeDAOJdbcImpl implements DocumentSearchResultsSizeDAO {
+    private static final String DOC_SEARCH_GENERATOR_SQL_PREFIX = "Select * ";
+    private static final String DOC_SEARCH_GENERATOR_REPLACEMENT_SQL_PREFIX = "Select count(*) ";
+    private static final org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(DocumentSearchResultsSizeDAOJdbcImpl.class);
+
+    private DataSource dataSource;
+    private DocumentSearchService documentSearchService;
+
+
+    @Override
+    public Long getTotalMatchingDocumentsSize(final DocumentSearchGenerator documentSearchGenerator, final DocumentSearchCriteria criteria, final DocumentType documentType) {

```

Comment:
Any reason why we need to include the documentType as a parameter to this method instead of passing in the List<RemotableAttributeField> searchFields? If we change the signature as mentioned, we can get rid of the DocumentSearchService dependency in the DAO..... (since we already have access to this service in DocumentSearchCriteriaBoLookupableHelperService and we can populate this list in here, instead of the DAO)